src/components/layouts/FilterBar.tsx

```diff
@@ -47,16 +44,41 @@ export const FilterBar = ({ mode, updateViewModeSetting }: FilterBarProps) => {
   const viewMode = viewSettingsTemp ? viewSettingsTemp.viewMode : view
   const viewModeFilterOptions = viewSettingsTemp ? (viewSettingsTemp.filterOptions as IFilterOptions) : filterOptions //ViewSettingsTemp used to apply temp values of viewSettings in filterOptions and viewMode because clientSideUpdate applies outdated cached values to original view and filterOptions if navigated
 
+  // Stores the initial assignee list for a particular filter type
+  const [initialAssignees, setInitialAssignees] = useState(filteredAssignee)
+  const [inputStatusValue, setInputStatusValue] = useState('')
+
   useEffect(() => {
     setFilteredAssignee(filteredAssigneeList)
   }, [filteredAssigneeList]) //to prevent filteredAssignee not updating when filteredAssigneeList changes when fetching assignee is delayed
 
+  useEffect(() => {
+    // Base these initial values off of first fetch of filteredAssignee
+    initialAssignees.length === 0 && setInitialAssignees(filteredAssignee)
+  }, [initialAssignees, filteredAssignee])
+
+  useEffect(() => {
+    // When focus is taken away from selector, make sure that assignee search results are replaced
+    if (filteredAssignee.length && initialAssignees.length && !inputStatusValue) {
+      setFilteredAssignee(initialAssignees)
+    }
+  }, [filteredAssignee, initialAssignees, inputStatusValue])
+
+  useEffect(() => {
+    if (!inputStatusValue) {
+      loading && setLoading(false)
+    }
+  }, [inputStatusValue])
```

Comment:
I think it is not a good idea to bring forward a lot of effects with similar dependency. Can we try to merge the logic into a single useEffect? I think the below suggestion would work - 

```javascript
useEffect(() => {
  // Base these initial values off of first fetch of filteredAssignee
  if (initialAssignees.length === 0) {
    setInitialAssignees(filteredAssignee);
  }

  // When focus is taken away from selector, reset filteredAssignee if needed
  if (filteredAssignee.length && initialAssignees.length && !inputStatusValue) {
    setFilteredAssignee(initialAssignees);
  }

  // Ensure loading state is reset if inputStatusValue is falsy
  if (!inputStatusValue && loading) {
    setLoading(false);
  }
}, [initialAssignees, filteredAssignee, inputStatusValue, loading]);

```